source/common/router/router.cc

```diff
@@ -147,14 +148,28 @@ void Filter::chargeUpstreamCode(const Http::HeaderMap& response_headers,
 
       Http::CodeUtility::chargeResponseStat(info);
     }
+
+    if (upstream_host) {
+      const uint64_t response_code = Http::Utility::getResponseStatus(info.response_headers_);
+      if (dropped) {
+        upstream_host->stats().rq_dropped_.inc();
+      } else if (response_code == 200) {
+        // TODO(htuch): When Envoy has first class support for gRPC on the data plane, we should
```

Comment:
We are already doing gRPC retry in the router using gRPC status codes, so I think we can just skip straight to doing this if we want.